spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java

```diff
@@ -82,7 +88,45 @@ protected void uploadOrRetrieveSkin(MinecraftProfileWrapper profile, Player play
             }
         }
     }
-
+    
+    public String[] RetrieveSkin(Player player) {
+
+        PlayerProfile playerProfile = player.getPlayerProfile();
+        if (playerProfile.hasTextures()) {
+            PropertyMap propertyMap = (PropertyMap) playerProfile.getProperties();
+            Property property = propertyMap.get("textures").iterator().next();
+            String texture = property.getValue();
+            String signature = property.getSignature();
+            
+            return new String[]{texture, signature};
+        }
+        
+        PlayerEntry playerEntry = database.getPlayerEntry(player.getUniqueId());
+        
+        SkinEntry setSkin = null;
+        long dateAdded = 0;
+        for (SkinEntry skinEntry : playerEntry.getSkinEntries()) {
+            if (dateAdded < skinEntry.getDateAdded()) {
+                dateAdded = skinEntry.getDateAdded();
+                setSkin = skinEntry;
+            }
+        }
+        if (setSkin != null) {
+            return new String[]{setSkin.getJavaSkinValue(), setSkin.getJavaSkinSignature()};
+        }
+        
+        /*
+            Default Steve Skin, Value and Signature
+            This is Provided if setSkin is Null
+            (Just to bypass any errors)
+        */
+        
+        String value = "ewogICJ0aW1lc3RhbXAiIDogMTYxMjY0NzA0MzEwOCwKICAicHJvZmlsZUlkIiA6ICJmYmYyMmIyMjNjN2E0NzA0OGYwM2U0MzVhZGFhNGVhZiIsCiAgInByb2ZpbGVOYW1lIiA6ICJCZWFyZGVkU3RldmUiLAogICJzaWduYXR1cmVSZXF1aXJlZCIgOiB0cnVlLAogICJ0ZXh0dXJlcyIgOiB7CiAgICAiU0tJTiIgOiB7CiAgICAgICJ1cmwiIDogImh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvMWE0YWY3MTg0NTVkNGFhYjUyOGU3YTYxZjg2ZmEyNWU2YTM2OWQxNzY4ZGNiMTNmN2RmMzE5YTcxM2ViODEwYiIKICAgIH0KICB9Cn0=";
```

Comment:
I think a cleaner implementation would just be to return null and let the plugin deal with it.